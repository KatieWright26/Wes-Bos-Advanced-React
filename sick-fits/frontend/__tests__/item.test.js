import { shallow } from 'enzyme';
import ItemComponent from '../components/Item';

const fakeItem = {
  id: 'BBC123',
  title: 'A Cool Item',
  description: 'A Rad Description',
  image: 'buyme.jpg',
  largeImage: 'BUYME.jpg',
  price: 6789,
};

describe('<Item/>', () => {
  it('renders and displays the price properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const PriceTag = wrapper.find('PriceTag');
    expect(PriceTag.children().text()).toBe('$67.89');
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  });
  it('renders the image properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const image = wrapper.find('img');
    expect(image.props().src).toBe(fakeItem.image);
    expect(image.props().alt).toBe(fakeItem.title);
  });
  it('renders out the buttons properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const buttonList = wrapper.find('.buttonList');
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link')).toHaveLength(1);
    expect(buttonList.find('AddToCart')).toHaveLength(1);
    expect(buttonList.find('DeleteItem')).toHaveLength(1);
  });
});
