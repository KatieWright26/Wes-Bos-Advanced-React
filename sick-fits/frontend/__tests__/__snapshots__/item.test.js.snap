// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="buyme.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "BBC123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $67.89
  </PriceTag>
  <p>
    A Rad Description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "BBC123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="BBC123"
    />
    <DeleteItem
      id="BBC123"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
